datafusion/optimizer/src/pre_cast_lit_in_comparison.rs

```diff
@@ -200,47 +160,110 @@ fn is_comparison_op(op: &Operator) -> bool {
 }
 
 fn is_support_data_type(data_type: &DataType) -> bool {
-    // TODO support decimal with other data type
     matches!(
         data_type,
-        DataType::Int8 | DataType::Int16 | DataType::Int32 | DataType::Int64
+        DataType::Int8
+            | DataType::Int16
+            | DataType::Int32
+            | DataType::Int64
+            | DataType::Decimal128(_, _)
     )
 }
 
-fn can_integer_literal_cast_to_type(
-    integer_lit_value: &ScalarValue,
+fn try_cast_literal_to_type(
+    lit_value: &ScalarValue,
     target_type: &DataType,
-) -> Result<bool> {
-    if integer_lit_value.is_null() {
+) -> Result<(bool, Option<ScalarValue>)> {
```

Comment:
It looks like this function always returns either `(true, Some(_))` or `(false, None)` so maybe it should just return the `Option` without the bool?

Reply:
good suggestion.
Done for your comments.
PTAL @andygrove